packages/jellyfish-api-core/__tests__/category/oracle/getFixedIntervalPrice.test.ts

```diff
@@ -0,0 +1,294 @@
+
+    // setLoanToken UBER
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'UBER',
+      fixedIntervalPriceId: 'UBER/USD'
+    })
+    await testing.generate(1)
+  }
+
+  it('should getFixedIntervalPrices', async () => {
+    {
+      const price = await testing.rpc.oracle.getFixedIntervalPrice('UBER/USD')
+      expect(price).toStrictEqual({
+        activePriceBlock: expect.any(Number),
+        nextPriceBlock: expect.any(Number),
+        fixedIntervalPriceId: 'UBER/USD',
+        activePrice: new BigNumber('0'),
+        nextPrice: new BigNumber('8'),
+        timestamp: expect.any(Number),
+        isLive: false
+      })
+    }
+
+    {
+      await testing.generate(6)
+      const price = await testing.rpc.oracle.getFixedIntervalPrice('UBER/USD')
+      expect(price).toStrictEqual({
+        activePriceBlock: expect.any(Number),
+        nextPriceBlock: expect.any(Number),
+        fixedIntervalPriceId: 'UBER/USD',
+        activePrice: new BigNumber('8'),
+        nextPrice: new BigNumber('8'),
+        timestamp: expect.any(Number),
+        isLive: true
+      })
+    }
+
+    {
+      await testing.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '32@UBER', currency: 'USD' }] })
+      const price = await testing.rpc.oracle.getFixedIntervalPrice('UBER/USD')
+      expect(price).toStrictEqual({
+        activePriceBlock: expect.any(Number),
+        nextPriceBlock: expect.any(Number),
+        fixedIntervalPriceId: 'UBER/USD',
+        activePrice: new BigNumber('8'),
+        nextPrice: new BigNumber('8'),
+        timestamp: expect.any(Number),
+        isLive: true
+      })
+    }
+
+    {
+      await testing.generate(6)
+      const price = await testing.rpc.oracle.getFixedIntervalPrice('UBER/USD')
+      expect(price).toStrictEqual({
+        activePriceBlock: expect.any(Number),
+        nextPriceBlock: expect.any(Number),
+        fixedIntervalPriceId: 'UBER/USD',
+        activePrice: new BigNumber('8'),
+        nextPrice: new BigNumber('32'),
+        timestamp: expect.any(Number),
+        isLive: false
+      })
+    }
+
+    {
+      await testing.generate(6)
+      const price = await testing.rpc.oracle.getFixedIntervalPrice('UBER/USD')
+      expect(price).toStrictEqual({
+        activePriceBlock: expect.any(Number),
+        nextPriceBlock: expect.any(Number),
+        fixedIntervalPriceId: 'UBER/USD',
+        activePrice: new BigNumber('32'),
+        nextPrice: new BigNumber('32'),
+        timestamp: expect.any(Number),
+        isLive: true
+      })
+    }
+  })
+
+  it('test prices changes within 6 blocks while updating price feed', async () => {
+    let activePriceBlock: number
+    let nextPriceBlock: number
+    let blockHeight: number
+    {
+      // make sure to have a valid price of 10@UBER
+      await testing.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '10@UBER', currency: 'USD' }] })
+      await testing.generate(13)
```

Comment:
Is there any way to know we need to have 13 blocks here?

Reply:
lemme move the ln:156 desc besides gen(13)

Reply:
I meant why this is not `14` blocks or `12` blocks? 
I asked this because, I really don't think there is any way to know the exact blocks we need to get the valid price
In my other PR, what I did is I wait until the next block which price become invalid, and then wait until the next block which price become valid back

Reply:
i got what your mean now
the price state is not affected by generated block, so using `waitForValidPrice` is more accurate